frontend/src/components/explorer/mapViewer/ThreeDviewer.vue

```diff
@@ -177,6 +181,13 @@ export default {
       }
     },
     async applyColorsAndRenderNetwork(levels) {
+      if (this.currentLevels === levels
+          || (this.currentLevels === {} && Object.keys(levels).length > 0)
+      ) {
+        return;
+      }
+      this.currentLevels = levels;
+
```

Comment:
I have a feeling this might change in the updated data overlay, so it doesn't matter so much how things are done now as long as they work.

Reply:
This was added because the [`EventBus`](https://github.com/MetabolicAtlas/MetabolicAtlas/blob/fix/dispose-3d-viewer/frontend/src/components/explorer/mapViewer/ThreeDviewer.vue#L87) was triggering `applyColorsAndRenderNetwork` too many times, which means the entire 3D viewer was being constructed multiple times. This takes up a lot of resources unnecessarily and it would also also make the disposing of things more cumbersome if we don't add this.